src/or/hs_common.c

```diff
@@ -1281,7 +1281,7 @@ node_has_hsdir_index(const node_t *node)
 
   /* A node can't have an HSDir index without a descriptor since we need desc
    * to get its ed25519 key */
-  if (!node_has_descriptor(node)) {
+  if (!node_has_preferred_descriptor(node, 0)) {
```

Comment:
Note: Using for_direct_connect = 0 is always correct for the HSDir hash ring, because we use non-bridge descriptors for the HSDir hash ring. This is true even though Tor2web connects directly to HSDirs.

Reply:
Hmm, does this need a comment?

Reply:
5398bf02fea4a5 has that comment.